platform/engineering/design-docs/2021-06-11-small-batch-publishing.md

```diff
@@ -0,0 +1,350 @@
+# **Small Batch Publishing**
+
+**Author(s):** Neil Hastings
+
+**Last Updated:** : June, 11 2021
+
+**Status:** **Draft** | In Review | Approved
+
+**Approvers:** 
+
+- [ ] Tim Wright 
+- [ ] Mike Chelen 
+- [ ] Michael Pritchard 
+- [ ] Dror Matalon
+- [ ] Steve Wirt
+- [ ] Elijah Lynn
+- [ ] VFS Public Website Team 
+- [ ] VFS Facilities Team
+
+## Table of Contents
+
+* [Overview](#overview)
+  * [Objective](#objective)
+  * [Links](#links)
+  * [Background](#background)
+  * [High Level Design](#high-level-design)
+* [Specifics](#specifics)
+  * [Detailed Design](#detailed-design)
+    * [Drupal Triggers Content Publishing](#drupal-triggers-content-publishing)
+    * [Preview Server Changes](#preview-server-changes)
+    * [Facility Banner Alerts](#facility-banner-alerts)
+    * [Metalsmith plugins](#metalsmith-plugins)
+    * [Publish Batch API](#publish-batch-api)
+    * [Continuous Integration](#continuous-integration)
+    * [Assets](assets)
+  * [Code Location](code-location)
+  * [Testing Plan](#testing-plan)
+  * [Logging](#logging)
+  * [Debugging](#debugging)
+  * [Caveats](#caveats)
+  * [Security Concerns](#security-concerns)
+  * [Privacy Concerns](#privacy-concerns)
+  * [Open Questions and Risks](#open-questions-and-risks)
+  * [Work Estimates](#work-estimates)
+  * [Alternatives, Research, and Future Work](#alternatives-Research-and-future-work)
+* [Long Term Vision](#long-term-vision)
+* [Revision History](#revision-history)
+
+## **Overview**
+
+
+### **Objective**
+
+An editor on the CMS should be able to publish content to va.gov when the "Publish" button is pressed in the CMS. This document outlines an approach utilizing the preview server’s code to render HTML which is directly deployed to S3.
+
+This design doc only covers publishing pages which are represented in Drupal as a node. The full content build still runs hourly.
```

Comment:
```suggestion
This design doc only covers publishing pages which are represented in Drupal as a node. The full content-only build still runs 9am-12pm Hourly, 1:45pm, 4pm, 5pm ET M-F.
```

Reply:
What does the content-only deploy contain if all the drupal content goes out immediately? The vagov-content stuff? If that's the case, I feel like we can just cancel the content-only deploys, that repo is barely ever updated

Reply:
We are planning on rolling out SBP by content type over a period of months.  After that, there are the pages from the markdown content repo and finally, pages (like listing and location pages) that use Drupal content to build a page but are not represented in Drupal.  Though https://github.com/department-of-veterans-affairs/va.gov-team/issues/25940 from @melissajschaff  might bring a solution.  Either a dependency graph in the build or moving the pages to Drupal would help with this use case.

Reply:
I think we should try and push for more unification of the content so we actually turn off the "hourly" content release. Seems like there would be a ton of times throughout the day were a build were happening that would block instant publishing making it a little more annoying rather than helpful

Reply:
I updated the plan to cutover content types from full build to SPB. This will make reduce lots of complexity.